### ISSUE TITLE
[Tracking] OKD 4.6 Stable

### ISSUE THREAD

**Issue by vrutkovs** [openshift/okd#348]
Features in this release:
* samples operator is using UBI / CentOS images
  Some imagestreams may be removed
* NetworkManager-ovs, glusterfs and open-vm-tools are being installed via OS extensions mechanism
  This ensures post-install actions are being run correctly on the machine. -ovs is required for OVN to function.
* Fedora 33 content

Blocking bugs:
* ~~[Bug 1882136 - created toolbox fails to start: open /proc/sys/net/ipv4/ping_group_range: Permission denied: OCI runtime permission denied error](https://bugzilla.redhat.com/show_bug.cgi?id=1882136) - blocks [4.6 machine-os-content promotion](https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/?job=promote-release-openshift-okd-machine-os-content-e2e-*-4.6)~~
  Fixed in F33
* [Bug 1888467 - Repos should be disabled in -firstboot.service before OS extensions are applied](https://bugzilla.redhat.com/show_bug.cgi?id=1888467)
	FCOS would use Fedora repos when OS extensions are being installed, may fail on disconnected installation
* ~~[4.5 -> 4.6 upgrade fails](https://bugzilla.redhat.com/show_bug.cgi?id=1880591) on CI due to OVN issues (needs verification, may be fixed by now)
  Needs [MCO PR](https://github.com/openshift/machine-config-operator/pull/2145) to land~~

Nice to have:
* [Bug 1844198 - the openvswitch service triggers SELinux denials on netlink_netfilter_socket ](https://bugzilla.redhat.com/show_bug.cgi?id=1844198)
* [Bug 1879099 - Extra reboot during 4.5 -> 4.6 upgrade](https://bugzilla.redhat.com/show_bug.cgi?id=1879099)
  This may prevent nodes to start during machine update (network is already 4.6, but nm-ovs is not yet installed) - needs verification, may be not necessary

TODO:
* [x] openshift/machine-config-operator#2199
      Fixes baremetal UPI install
* [x] openshift/machine-config-operator#2200
      release-4.6 cherry-pick of ^
* [x] openshift/machine-config-operator#2196
      Handle kubelet proxy config from plain file to a systemd dropin in Ignition
* [x] openshift/machine-config-operator#2246
      release-4.6 cherry-pick of ^
* [x] https://github.com/openshift/okd/issues/352
      Incorrect hostname set (GCP only)
* [x] Static IP config with nameserver is broken due to systemd-resolved
      https://github.com/openshift/okd/issues/380. Workaround: initial FCOS image should be F32

Post-release todolist:
* [Move manifests installed by installer to okd-machine-os manifest](https://github.com/openshift/okd/issues/382)
* [Update installer and okd-machine-os to use stable FCOS once F33 content lands there](https://github.com/openshift/okd/issues/383)
* ~~[CRI-O 1.19](https://github.com/openshift/okd/issues/341)~~
* samples operator is using UBI / CentOS images - compile a list of differences

**Comment by vrutkovs** [comment]
State as of Oct 15:

Managed to update 4.5 to custom 4.6 with the following PRs
* [ ] https://github.com/openshift/machine-config-operator/pull/2161
* [ ] https://github.com/openshift/ovn-kubernetes/pull/307
* [ ] https://github.com/openshift/machine-config-operator/pull/2154
* [ ] https://github.com/openshift/release/pull/12683
* [ ] https://github.com/openshift/machine-config-operator/pull/2160
* [ ] machine-os-content with MC to add `extensions` on upgrade (https://github.com/vrutkovs/machine-os-content-builder/tree/devel/manifests)

**Comment by vrutkovs** [comment]
Remaining list of PRs to merge:
* [ ] https://github.com/openshift/machine-config-operator/pull/2199
      Fixes baremetal UPI install
* [ ] https://github.com/openshift/machine-config-operator/pull/2200
      release-4.6 cherry-pick of ^
* [ ] https://github.com/openshift/machine-config-operator/pull/2198
      Handle kubelet proxy config from plain file to a systemd dropin in Ignition
* [ ] https://github.com/coreos/fedora-coreos-tracker/issues/649
      Incorrect hostname set (GCP only)

**Comment by LorbusChris** [comment]
We might need https://github.com/openshift/machine-config-operator/pull/2160 as a temporary fix after all until https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/issues/572 is implemented

**Comment by willmtemple** [comment]
> samples operator is using UBI / CentOS images
> Some imagestreams may be removed

I manually one-time provisioned the okd-x86_64 Templates and CentOS-based imagestreams from the operator into the `openshift` namespace on my OKD 4.5.0 lab cluster, will there be any action during the update for me or will the operator simply take over managing them?

Really looking forward to this update.

**Comment by vrutkovs** [comment]
>will there be any action during the update for me or will the operator simply take over managing them?

The operator will replace the managed (like `python`, `wildfly` etc.). The rest will be untouched

**Comment by jomeier** [comment]
It looks like that almost all open tasks are resolved 👍

**Comment by LorbusChris** [comment]
Update

Open Issues:
- [ ] https://github.com/openshift/machine-config-operator/pull/2217
- [ ] https://github.com/openshift/okd/issues/380

Pending Backports:
- [ ] https://github.com/openshift/machine-config-operator/pull/2161
- [ ] https://github.com/openshift/machine-config-operator/pull/2245
- [ ] https://github.com/openshift/machine-config-operator/pull/2246
- [ ] Backport of https://github.com/openshift/machine-config-operator/pull/2217

**Comment by bdurrow** [comment]
Will 4.6 RC hold for FCOS 33 to land in stable or would that only be a requirement for GA?  If so it seems that will be the critical path.

**Comment by Conan-Kudo** [comment]
There's also an issue where Fedora CoreOS has the wrong `iptables` implementation: https://github.com/coreos/fedora-coreos-tracker/issues/676

**Comment by LorbusChris** [comment]
> Will 4.6 RC hold for FCOS 33 to land in stable or would that only be a requirement for GA? If so it seems that will be the critical path.

We won't hold RC for that

> There's also an issue where Fedora CoreOS has the wrong iptables implementation: coreos/fedora-coreos-tracker#676

I'm not sure that's actually an issue for OKD - in 4.5 it works with the older implementation. Still a good idea to track progress on it here, too :)

**Comment by dustymabe** [comment]
> > There's also an issue where Fedora CoreOS has the wrong iptables implementation: [coreos/fedora-coreos-tracker#676](https://github.com/coreos/fedora-coreos-tracker/issues/676)
> 
> I'm not sure that's actually an issue for OKD - in 4.5 it works with the older implementation. Still a good idea to track progress on it here, too :)

The complicated part is if you start 4.6 with iptables legacy and then it switches to nft. We're going to try to make sure existing nodes don't get switched on upgrade, but new nodes would have nft. Just something to consider.

**Comment by ibotty** [comment]
Why is it a problem to switch? Can't it be switched to nft when the node gets started with the new compose?  What's the failure case?

**Comment by dustymabe** [comment]
Maybe it's no problem. I just don't know. It looks like we're going to hold off for a while on the migration so just assume iptables legacy for now (which is working fine).

**Comment by vrutkovs** [comment]
https://amd64.origin.releases.ci.openshift.org/releasestream/4-stable/release/4.6.0-0.okd-2020-11-27-200126
https://github.com/openshift/okd/releases/tag/4.6.0-0.okd-2020-11-27-200126

**Comment by titou10titou10** [comment]
Is there a migration path from 4.5.0-0.okd-2020-10-15-235428 to 4.6.0-0.okd-xxx ?

**Comment by vrutkovs** [comment]
>Is there a migration path from 4.5.0-0.okd-2020-10-15-235428 to 4.6.0-0.okd-xxx ?

Not yet, should be available shortly once tests finish. See "Upgrades from" section in https://amd64.origin.releases.ci.openshift.org/releasestream/4-stable/release/4.6.0-0.okd-2020-11-27-200126

**Comment by marshallford** [comment]
> Not yet, should be available shortly once tests finish. See "Upgrades from" section in https://amd64.origin.releases.ci.openshift.org/releasestream/4-stable/release/4.6.0-0.okd-2020-11-27-200126

I'm getting:

> Unable to show full changelog: could not generate a changelog: error: Could not load commits for https://github.com/openshift/installer: from commit 63200c80c431b8dbaa06c0cc13282d819bd7e5f8 does not exist
